server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -486,6 +487,27 @@ tableName, getSerialType(), getPayloadType()
     );
   }
 
+  public void createSegmentSchemaTable(final String tableName)
```

Comment:
wouldn't there be two tables instead?

Reply:
What is the second table for?
I have added a table for segment schema and added `schema_id` column to dataSegments table